### DISCUSSION TITLE
不知道改如何监听事件

### DISCUSSION THREAD

**Discussion by NoNameGMM** [LagrangeDev/Lagrange.Core#338] [Q&A] [answered]
EventInvoker eventInvoker = new EventInvoker(bot);
提示“EventInvoker”不包含采用 1 个参数的构造函数

**Comment by ghost** [comment, marked as answer]
参考 [Lagrange.OneBot/Core/Notify/NotifyService.cs#L13](https://github.com/KonataDev/Lagrange.Core/blob/a720deef1806138688000d43c2e72efd1a246911/Lagrange.OneBot/Core/Notify/NotifyService.cs#L13)

**Reply by NoNameGMM** [reply]
直接...

bot.Invoker.OnBotOnlineEvent += (_, @event) =>
{
    Console.WriteLine("机器人已上线");
};

这样吗

**Reply by ghost** [reply]
对

**Reply by NoNameGMM** [reply]
```csharp
await bot.LoginByQrCode();
while (true)
{
    bot.Invoker.OnBotOnlineEvent += (_, @event) =>
    {
        Console.WriteLine("机器人已上线");
    };
}
```

这样...没输出

**Reply by NoNameGMM** [reply]
我该..怎么改啊

**Reply by ghost** [reply]
额...你可能需要查看一下事件模型是怎么运作的

你应该要先订阅事件再登陆, 订阅事件不需要 while, 只需要订阅一次就行, 事件到达会自动运行这个方法

```csharp
bot.Invoker.OnBotOnlineEvent += (_,@event) => { Console.WriteLine("机器人已上线"); };

await bot.LoginByQRCode();
```

**Reply by NoNameGMM** [reply]
啊抱歉..我懂了
谢谢
